docs/general/dql/selects.rst

```diff
@@ -740,6 +740,43 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records having values with empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (15, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty arrays, is::
+
+    cr> select count(*) from locations where
+    ... landmarks is null or
+    ... landmarks = [];
+    +----------+
+    | count(*) |
+    +----------+
+    |       14 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+For determining the number of items in an array, you can use the
+:ref:`array_length <scalar-array_length>` scalar function::
+
+    cr> select count(*) from locations where
+    ... array_length(landmarks, 1) = 2;
+    +----------+
+    | count(*) |
+    +----------+
+    |        1 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+
+
```

Comment:
Also here we could drop new values to avoid that following result sets must be adjusted.

Reply:
I was thinking about it as well. If you think dropping the record within a hidden section is the right way to minimize the impact on subsequent tests, I will be happy to amend this patch that way.

Reply:
Not sure, but I don't like that this additional documentation requires us to adjust follow up documentation. So +1 from my side for dropping it.

Reply:
Thanks. Let's see what @mfussenegger has to say about it.